src/main/java/uk/gov/ida/verifyserviceprovider/services/UnifyingKeySignatureTrustEngine.java

```diff
@@ -0,0 +1,105 @@
+package uk.gov.ida.verifyserviceprovider.services;
+
+import com.google.common.collect.Lists;
+import net.shibboleth.utilities.java.support.resolver.CriteriaSet;
+import net.shibboleth.utilities.java.support.resolver.ResolverException;
+import org.opensaml.security.credential.Credential;
+import org.opensaml.security.credential.CredentialResolver;
+import org.opensaml.xmlsec.keyinfo.KeyInfoCredentialResolver;
+import org.opensaml.xmlsec.signature.support.impl.ExplicitKeySignatureTrustEngine;
+import uk.gov.ida.saml.metadata.EidasMetadataResolverRepository;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.LinkedList;
+import java.util.List;
+
+public class UnifyingKeySignatureTrustEngine extends ExplicitKeySignatureTrustEngine {
```

Comment:
It may be better to call it `VerifyAndEidasKeySignatureTrustEngine`. I think `Unifying` is unclear out of context - what exactly is this trust engine unifying?